cryptoconditions/crypto.py

```diff
@@ -1,269 +1,121 @@
 import base64
-
 import base58
-import ed25519
-
-
-class SigningKey(metaclass=ABCMeta):
-    """
-    SigningKey instance
-    """
-
-    @abstractmethod
-    def sign(self, data):
-        """
-        Sign data with private key
+import nacl.signing
+import nacl.encoding
+import nacl.exceptions
 
-        Args:
-            data:
-        """
+from cryptoconditions import exceptions
 
-    @abstractmethod
-    def get_verifying_key(self):
-        """
-        Get the associated verifying key
-
-        Returns:
-            A VerifyingKey object
-        """
-
-    @abstractmethod
-    def to_ascii(self, prefix, encoding):
-        """
-        Encode the external value
-
-        Args:
-            prefix:
-            encoding:
-        """
-
-    @staticmethod
-    @abstractmethod
-    def encode(private_value):
-        """
-        Encode the internal private_value to base58
 
-        Args:
-            private_value:
-        """
+class Base58Encoder(object):
 
     @staticmethod
-    @abstractmethod
-    def decode(private_base58):
-        """
-        Decode the base58 private value to internal value
-
-        Args:
-            private_base58 (base58):
-        """
-        raise NotImplementedError
-
-
-class VerifyingKey(metaclass=ABCMeta):
-
-    @abstractmethod
-    def verify(self, data, signature):
-        """
-        Check the if the signature matches the data and this verifyingkey
-
-        Args:
-            data:
-            signature:
-
-        Returns:
-            boolean:
-        """
-
-    @abstractmethod
-    def to_ascii(self, prefix, encoding):
-        """
-        Encode the external value
-
-        Args:
-            prefix:
-            encoding:
-        """
+    def encode(data):
+        return base58.b58encode(data).encode()
 
     @staticmethod
-    @abstractmethod
-    def encode(public_value):
-        """
-        Encode the public key to base58 represented by the internal values
-
-        Args:
-            public_value
-        """
-
-    @staticmethod
-    @abstractmethod
-    def decode(public_base58):
-        """
-        Decode the base58 public_value to internal value
-
-        Args:
-            public_base58 (base58):
-        """
-
-
-class Ed25519SigningKey(ed25519.SigningKey, SigningKey):
+    def decode(data):
+        return base58.b58decode(data)
+
+
+def _get_nacl_encoder(encoding):
+    if encoding == 'base58':
+        return Base58Encoder
+    elif encoding == 'base64':
+        return nacl.encoding.Base64Encoder
+    elif encoding == 'base32':
+        return nacl.encoding.Base32Encoder
+    elif encoding == 'base16':
+        return nacl.encoding.Base16Encoder
+    elif encoding == 'hex':
+        return nacl.encoding.HexEncoder
+    elif encoding is 'bytes':
+        return nacl.encoding.RawEncoder
+    else:
+        raise exceptions.UnknownEncodingError("Unknown or unsupported encoding")
+
+
+class Ed25519SigningKey(nacl.signing.SigningKey):
```

Comment:
Given that we don't use the `generate()` classmethod (and it probably doesn't work?), perhaps composition here would be preferable to inheritance?